app/sections/collection-header.tsx

```diff
@@ -0,0 +1,142 @@
+import type {
+    HydrogenComponentProps,
+    HydrogenComponentSchema,
+} from '@weaverse/hydrogen';
+import { forwardRef } from 'react';
+import { useLoaderData } from '@remix-run/react';
+import { CollectionDetailsQuery } from 'storefrontapi.generated';
+
+interface HeaderProps extends HydrogenComponentProps {
+    heightSectionDesktop: number;
+    heightSectionMobile: number;
+    enableBackground: boolean;
+    textPosition: string;
+    overlayBackground: number;
+    enableOverlay: boolean;
+}
+
+let CollectHeader = forwardRef<HTMLElement, HeaderProps>((props, ref) => {
+    let { heightSectionDesktop, heightSectionMobile, enableBackground, textPosition, overlayBackground, enableOverlay, ...rest } = props;
+    let { collection } = useLoaderData<
+        CollectionDetailsQuery & {
+            collections: Array<{ handle: string; title: string }>;
+        }
+    >();
+    let backgroundStyle = {};
+    if (enableBackground) {
+        backgroundStyle = {
+            backgroundImage: `url('${collection?.image?.url}')`,
+            '--header-height-desktop': `${heightSectionDesktop}px`,
+            '--header-height-mobile': `${heightSectionMobile}px`
+        };
+    }
+    let textStyle = {};
+    if (enableBackground) {
+        textStyle = {
+            color: '#fff',
+        };
+    } else {
+        textStyle = {
+            color: '#333',
+            position: 'relative',
+        };
+    }
+    let overlayStyle = {};
+    if (enableOverlay && enableBackground) {
+        overlayStyle = {
+            backgroundColor: `rgba(0, 0, 0, ${overlayBackground})`,
+        };
+    }
+    return (
+        <section ref={ref} {...rest} style={{ ...backgroundStyle }} className={`flex items-center justify-start relative overflow-hidden bg-center bg-no-repeat bg-cover h-[var(--header-height-mobile)] sm:h-[var(--header-height-desktop)]`}>
+            {enableOverlay && enableBackground && <div className="absolute inset-0 z-1" style={overlayStyle}></div>}
+            <div className={`text-center mx-auto w-5/6 text-gray-700 ml-auto absolute ${textPosition} z-2`} style={textStyle}>
```

Comment:
Don't use template literal to add a class name, use [clsx](https://www.npmjs.com/package/clsx) instead (it's been already installed in Pilot)